warehouse/packaging/tasks.py

```diff
@@ -23,10 +24,38 @@
 from warehouse.accounts.models import User, WebAuthn
 from warehouse.email import send_two_factor_mandate_email
 from warehouse.metrics import IMetricsService
+from warehouse.packaging.interfaces import IFileStorage
 from warehouse.packaging.models import Description, File, Project, Release, Role
 from warehouse.utils import readme
 
 
+@tasks.task(ignore_result=True, acks_late=True)
+def sync_file_to_archive(request, file_id):
+    file = request.db.query(File).get(file_id)
+    if not file.archived:
+        primary_storage = request.find_service(IFileStorage, name="primary")
+        archive_storage = request.find_service(IFileStorage, name="archive")
+        metadata = primary_storage.get_metadata(file.path)
+        file_obj = primary_storage.get(file.path)
+        with tempfile.NamedTemporaryFile() as file_for_archive:
+            file_for_archive.write(file_obj.read())
+            file_for_archive.flush()
+            archive_storage.store(file.path, file_for_archive.name, meta=metadata)
+        file.archived = True
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏does setting the property value like this run the relevant SQL `UPDATE` or is there some other `.commit()` thingie we have to call?

Reply:
transaction manager thing-a-ma-bob does the commits for us at the end of the "request" (we're in celery so that's not a real request).

that's as much as I know, @dstufft could probably expound.